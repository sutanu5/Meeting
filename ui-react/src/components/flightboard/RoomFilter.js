import React, { Component } from 'react';
import PropTypes from 'prop-types';
import * as config from '../../config/flightboard.config.js';

class RoomFilter extends Component {

  filterFlightboard = (e) => {
    this.props.filter(e.target.id);
  }

  render() {
    const { error, response, roomlists } = this.props;

    return (
      <li>
        <a href="#" className="current-filter">
          {config.roomFilter.filterTitle}
        </a>
        <ul className="menu fb__child-dropdown">
          <li onClick={this.filterFlightboard} id="roomlist-all">
            {config.roomFilter.filterAllTitle}
          </li>

          { response && !error ?
            roomlists.map((item, key) =>
              <li onClick={this.filterFlightboard} key={key} id={'roomlist-' + item.toLowerCase().replace(/\s+/g, "-")}>
                {item}
              </li>
            )
          :
            <li id="roomlist__loading">
              Loading ...
            </li>
          }

        </ul>
      </li>
    );
  }
}

RoomFilter.propTypes = {
  filter: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.string
  ])
};

export default RoomFilter;
