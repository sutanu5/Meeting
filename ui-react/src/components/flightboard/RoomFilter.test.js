import RoomFilter from './RoomFilter';

describe('Flightboard RoomFilter', () => {
  let mockFilter, props, roomlists;

  beforeEach(() => {
    mockFilter = sinon.spy();
    props = {
      response: true,
      error: false,
      roomlists: [
        'Roomlist Number One',
        'Roomlist Number Two'
      ]
    }
  });

  // filterTitle - test .current-filter
  // filterAllTitle - roomlist-all


  it('renders correctly', () => {
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders filterTitle', () => {
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);

    const filterTitle = wrapper.find('.current-filter').text();
    expect(filterTitle).not.toBeEmpty();
  });

  it('renders filterAllTitle', () => {
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);

    const filterAllTitle = wrapper.find('#roomlist-all').text();
    expect(filterAllTitle).not.toBeEmpty();
  });

  it('renders "loading" in list if response is false and error is false', () => {
    props.response = false;
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);

    const loading = wrapper.find('#roomlist__loading').exists();
    expect(loading).toBeTruthy();
  });

  it('renders "loading" in list if response is true and error is true', () => {
    props.error = true;
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);

    const loading = wrapper.find('#roomlist__loading').exists();
    expect(loading).toBeTruthy();
  });

  it('renders "loading" in list if response is false and error is true', () => {
    props.error = true;
    props.response = false;
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);

    const loading = wrapper.find('#roomlist__loading').exists();
    expect(loading).toBeTruthy();
  });

  it('does not render "loading" in list if response is true and error is false', () => {
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);

    const loading = wrapper.find('#roomlist__loading').exists();
    expect(loading).toBeFalsy();
  });

  it('list renders with three children', () => {
    const wrapper = shallow(<RoomFilter {...props} filter={mockFilter} />);
    const listItems = wrapper.find('.fb__child-dropdown').children();

    expect(listItems).toHaveLength(3);
  });

  it('operates filter callback function correctly when clicked', () => {
    const wrapper = mount(<RoomFilter {...props} filter={mockFilter} />);
    wrapper.find('#roomlist-all').simulate('click', {target: {id: 'foo'}});

    expect(mockFilter.calledOnce).toBe(true);
  });
});
