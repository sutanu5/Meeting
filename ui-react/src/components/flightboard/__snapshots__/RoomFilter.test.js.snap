// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flightboard RoomFilter renders correctly 1`] = `
<li>
  <a
    className="current-filter"
    href="#"
  >
    Locations
  </a>
  <ul
    className="menu fb__child-dropdown"
  >
    <li
      id="roomlist-all"
      onClick={[Function]}
    >
      All Conference Rooms
    </li>
    <li
      id="roomlist-roomlist-number-one"
      key="0"
      onClick={[Function]}
    >
      Roomlist Number One
    </li>
    <li
      id="roomlist-roomlist-number-two"
      key="1"
      onClick={[Function]}
    >
      Roomlist Number Two
    </li>
  </ul>
</li>
`;
